numpy/core/numeric.py

```diff
@@ -166,7 +176,7 @@ def zeros_like(a, dtype=None, order='K', subok=True):
     array([0.,  0.,  0.])
 
     """
-    res = empty_like(a, dtype=dtype, order=order, subok=subok)
+    res = empty_like(a, dtype=dtype, order=order, subok=subok, shape=shape)
     # needed instead of a 0 to get same result as zeros for for string dtypes
     z = zeros(1, dtype=res.dtype)
     multiarray.copyto(res, z, casting='unsafe')
```

Comment:
Is the `casting=unsafe` needed, given that this is already the same `dtype`?

Reply:
I honestly have no idea, this isn't part of this PR and I'd rather not change things out of its context.

Reply:
Fair enough. 👍